# Coursera Capstone Final Project 
## The Battle of Neighborhoods (Week 1)
### Mariano Puga - Part 1 : Introduction and Data Sections

### We import the libraries

In [1]:
import pandas as pd
import numpy as np
import json

import requests
from pandas.io.json import json_normalize

import matplotlib.cm as cm
import matplotlib.colors as colors

from sklearn.cluster import KMeans

!pip install geopy
from geopy.geocoders import Nominatim

!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

Solving environment: done

# All requested packages already installed.

Libraries imported.


## Coursera Capstone - REPORT CONTENT
1. Introduction Section : ⁃ Discussion of the business problem and the interested audience in this project.
2. Data Section:
⁃ Description of the data that will be used to solve the problem and the sources.
3. Methodology section: 
⁃ Discussion and description of exploratory data analysis carried out, any inferential statistical testing performed, and if any machine learnings were used establishing the strategy and purposes.
4. Results section: 
⁃ Discussion of the results.
5. Discussion section: 
⁃ Elaboration and discussion on any observations noted and any recommendations suggested based on the results.
6. Conclusion section: 
⁃ Report Conclusion.

### A description of the problem and a discussion of the background. (15 marks)


# 1. Introduction Section :

## Discussion of the business problem and the audience who would be interested in this project.
### Description of the Problem and Background
**Scenario:**

I'm currently working in Santiago (Chile, a small country in south america for those who don't know), but I wanna work overseas to get a hold of deeper experiences and develop new abilities. Also, I'm a huge surfing fan so I'd love to combine a city big enough to study an MBA and also stay close to the ocean.   
The first place I thought of in order to do this was Melbourne, Australia. I want new experiences so I wont compare my current situation (venues I tend to go to for example) to my new prospects but I'd love to find a place with the following amenities:

- Residential Apartment in order to share it (hopefully 3 bedrooms)
- Easy access to the surf spots.
- Desirable to have venues such as coffee shops, Asian restaurants and gym's
- Rent shouldn't be too high, I found most of my information on https://www.kaggle.com/anthonypino/melbourne-housing-market and there wasn't information on rent prices so I'll use housing prices to get an idea of how expensive is each suburb.  

#### Business Problem:
The challenge is to find a suitable suburb to live in in Melbourne that complies with the demands on location, price and venues. The data required to resolve this challenge is described in the following section 2, below.

#### Interested Audience
I'm particulary interested in surfing so I guess this is oriented to those recent graduates that want to explore studying/working close to the ocean 

### A description of the data and how it will be used to solve the problem. (15 marks)


# 2. Data Section:

## Description of the data and its sources that will be used to solve the problem

### Description of the Data:

The following data is required to answer the issues of the problem:

1. List of suburbs of Melbourne with their geodata (latitud and longitud).
2. List of housing prices for each suburb.
3. Venues for each of Melbourne's neighborhoods.


## How the data will be used to solve the problem
### The data will be used as follows:

- Use Foursquare and geopy data to map top 10 venues for all Melbourne neighborhoods and clustered in groups 
- Evaluate housing in Melbourne for each suburb.
- Addresses from posible locations will be converted to geodata( lat, long) using Geopy-distance and Nominatim.
- Data will be searched in open data sources if available, from real estate sites if open to reading, libraries or other government agencies.

(https://www.kaggle.com/anthonypino/melbourne-housing-market)
(https://data.melbourne.vic.gov.au/)


### The procesing of these DATA will allow to answer the key questions to make a decision:

- Which suburb is more expensive?
- What is the area of Melbourne with best rental pricing that meets criteria established? (estimated based on house prices)
- What is the distance from the ocean and the tentative future home?
- How do the prices compare?
- Can I have a really good coffee just before I go surfing?
- Are there tradeoffs between size and price and location?
- Any other interesting statistical data findings of the real estate and overall data.


### First we upload de diferent suburbs of Melbourne (we use this table only to check if we have all our suburbs in df2)

In [18]:
url1 = 'https://raw.githubusercontent.com/marianopuga/Battle-of-the-neighborhoods/master/Melbourne.xls'
read_table = pd.read_csv(url1,header=[0])
df1 = read_table
df1

City     Neighborhood
0   Melbourne          Carlton
1   Melbourne    Carlton North
2   Melbourne        Docklands
3   Melbourne   East Melbourne
4   Melbourne       Flemington
5   Melbourne       Kensington
6   Melbourne   Melbourne 3000
7   Melbourne   Melbourne 3004
8   Melbourne  North Melbourne
9   Melbourne   Port Melbourne
10  Melbourne        Parkville
11  Melbourne      South Wharf
12  Melbourne      South Yarra
13  Melbourne        Southbank
14  Melbourne   West Melbourne

### Then we upload the coordinates for each suburb

In [19]:
url2 = 'https://raw.githubusercontent.com/marianopuga/Battle-of-the-neighborhoods/master/Address_Points%20(1).csv'
read_table = pd.read_csv(url2,header=[0])
df2 = read_table
df2

StreetNo  GISID                                   the_geom  \
0           84  28064  POINT (144.979502658185 -37.835383325166)   
1          261  26268  POINT (144.984359089201 -37.848932002343)   
2          129  57067  POINT (144.966043133987 -37.814669852174)   
3            1  20179   POINT (144.96875067939 -37.810206322362)   
4           26   9563  POINT (144.977804675755 -37.817028046999)   
5           29  54207  POINT (144.961846725915 -37.823294521978)   
6          495  10692  POINT (144.974179390899 -37.809519854086)   
7           20  35522  POINT (144.945620594716 -37.794806537703)   
8           54  28401  POINT (144.984642530147 -37.842417064547)   
9            9  62712  POINT (144.942256563844 -37.805990926179)   
10         191   3933  POINT (144.968606218146 -37.796217998126)   
11          32   2043  POINT (144.965839334856 -37.801181589513)   
12         122  62591   POINT (144.943500316856 -37.80502026006)   
13         110  57505  POINT (144.936905628998 -37.796576733767)   
14          61    482  POINT (144.968271133959 -37.796461715388)   
15           7  61703  POINT (144.930628016834 -37.790587789058)   
16         242   1718   POINT (144.960808940581 -37.80415609047)   
17         611  30458  POINT (144.947032065761 -37.802908194833)   
18         425  16407   POINT (144.96055707709 -37.808975676046)   
19          62  62099  POINT (144.943284693461 -37.821879581249)   
20          41  62369    POINT (144.9159755063 -37.829907372716)   
21          20  57755  POINT (144.923850314205 -37.788384721704)   
22         372  10200   POINT (144.97882453527 -37.809754022957)   
23         113  33183  POINT (144.943154290803 -37.803878210616)   
24         348  59089  POINT (144.959635019771 -37.785504493426)   
25          19  53662  POINT (144.947682653637 -37.794625677938)   
26          56  27504  POINT (144.980379490579 -37.836036543356)   
27          43  35269   POINT (144.942835552775 -37.79238465343)   
28         326  15913  POINT (144.956489958468 -37.811520880515)   
29          44  24700  POINT (144.954786907433 -37.795603140044)   
...        ...    ...                                        ...   
63624      197  23335   POINT (144.965232982607 -37.81305727474)   
63625       81  50878   POINT (144.95525519269 -37.796943108707)   
63626      293  21039  POINT (144.969911619897 -37.799864665938)   
63627     1083   4559  POINT (144.990006836247 -37.816340691708)   
63628      405  30288  POINT (144.954234589776 -37.803702386449)   
63629        4  17479  POINT (144.972752063547 -37.811296105038)   
63630        4  61990  POINT (144.978409577725 -37.835088696751)   
63631       33   6113   POINT (144.985175387531 -37.81545801779)   
63632      119  61972  POINT (144.978621334623 -37.834491215601)   
63633       32  34357  POINT (144.941335605428 -37.804397039278)   
63634      219  13586  POINT (144.962776103128 -37.810047902108)   
63635        2   4158  POINT (144.969936897742 -37.799280198839)   
63636      426  16423  POINT (144.960844491744 -37.808797532802)   
63637       30  34822  POINT (144.948884191096 -37.797392859868)   
63638       37  32082  POINT (144.957283499826 -37.804400901781)   
63639      114  33548   POINT (144.94859126258 -37.801307510798)   
63640       70  59283  POINT (144.973121664615 -37.797387151576)   
63641       72  52294  POINT (144.933828168116 -37.792463878477)   
63642       25  20016  POINT (144.964319494394 -37.806121107088)   
63643      143  40200  POINT (144.963046125358 -37.828280321548)   
63644       24  60704   POINT (144.954158690294 -37.81515227964)   
63645       61  44527  POINT (144.919923177076 -37.796770296811)   
63646      485  40407  POINT (144.959462428005 -37.781386781115)   
63647       71  62357  POINT (144.918920468096 -37.828588176643)   
63648      532  18462  POINT (144.957900721104 -37.815621683326)   
63649       18  11336   POINT (144.94007925619 -37.824496832489)   
63650      292  40174  POINT (144.971055733073 -37.82777386468

### Drop unnecesary Data

In [20]:
df2.drop(['GISID','the_geom','AddComp','EASTING','NORTHING','AddressPnt','StrName','StreetNo','StreetID'], axis = 1, inplace = True)
df2

Suburb  SuburbID   LATITUDE   LONGITUDE
0          South Yarra     595.0 -37.835383  144.979503
1            Melbourne     591.0 -37.848932  144.984359
2            Melbourne     591.0 -37.814670  144.966043
3            Melbourne     591.0 -37.810206  144.968751
4       East Melbourne     587.0 -37.817028  144.977805
5            Southbank     596.0 -37.823295  144.961847
6       East Melbourne     587.0 -37.809520  144.974179
7      North Melbourne     592.0 -37.794807  144.945621
8          South Yarra     595.0 -37.842417  144.984643
9       West Melbourne     597.0 -37.805991  144.942257
10             Carlton     585.0 -37.796218  144.968606
11             Carlton     585.0 -37.801182  144.965839
12      West Melbourne     597.0 -37.805020  144.943500
13     North Melbourne     592.0 -37.796577  144.936906
14             Carlton     585.0 -37.796462  144.968271
15          Kensington     590.0 -37.790588  144.930628
16             Carlton     585.0 -37.804156  144.960809
17     North Melbourne     592.0 -37.802908  144.947032
18           Melbourne     591.0 -37.808976  144.960557
19           Docklands     599.0 -37.821880  144.943285
20      Port Melbourne     594.0 -37.829907  144.915976
21          Kensington     590.0 -37.788385  144.923850
22      East Melbourne     587.0 -37.809754  144.978825
23     North Melbourne     592.0 -37.803878  144.943154
24       Carlton North     586.0 -37.785504  144.959635
25     North Melbourne     592.0 -37.794626  144.947683
26         South Yarra     595.0 -37.836037  144.980379
27     North Melbourne     592.0 -37.792385  144.942836
28           Melbourne     591.0 -37.811521  144.956490
29           Parkville     593.0 -37.795603  144.954787
...                ...       ...        ...         ...
63624        Melbourne     591.0 -37.813057  144.965233
63625        Parkville     593.0 -37.796943  144.955255
63626          Carlton     585.0 -37.799865  144.969912
63627   East Melbourne     587.0 -37.816341  144.990007
63628  North Melbourne     592.0 -37.803702  144.954235
63629        Melbourne     591.0 -37.811296  144.972752
63630        Melbourne     591.0 -37.835089  144.978410
63631   East Melbourne     587.0 -37.815458  144.985175
63632        Melbourne     591.0 -37.834491  144.978621
63633   West Melbourne     597.0 -37.804397  144.941336
63634        Melbourne     591.0 -37.810048  144.962776
63635          Carlton     585.0 -37.799280  144.969937
63636        Melbourne     591.0 -37.808798  144.960844
63637  North Melbourne     592.0 -37.797393  144.948884
63638  North Melbourne     592.0 -37.804401  144.957283
63639  North Melbourne     592.0 -37.801308  144.948591
63640          Carlton     585.0 -37.797387  144.973122
63641       Kensington     590.0 -37.792464  144.933828
63642          Carlton     585.0 -37.806121  144.964319
63643        Southbank     596.0 -37.828280  144.963046
63644        Melbourne     591.0 -37.815152  144.954159
63645       Kensington     590.0 -37.796770  144.919923
63646        Parkville     593.0 -37.781387  144.959462
63647   Port Melbourne     594.0 -37.828588  144.918920
63648        Melbourne     591.0 -37.815622  144.957901
63649        Docklands     599.0 -37.824497  144.940079
63650        Southbank     596.0 -37.827774  144.971056
63651              NaN       0.0 -37.781334  144.910085
63652   West Melbourne     597.0 -37.812042  144.949112
63653      South Yarra     595.0 -37.837952  144.982824

[63654 rows x 4 columns]

### Now we group 

In [75]:
Melbourne = df2.groupby('Suburb').mean().reset_index()
Melbourne

Suburb  SuburbID   LATITUDE   LONGITUDE
0           Carlton     585.0 -37.799815  144.967612
1     Carlton North     586.0 -37.783667  144.960043
2         Docklands     599.0 -37.818229  144.944143
3    East Melbourne     587.0 -37.813617  144.983546
4        Flemington     588.0 -37.787583  144.918486
5        Kensington     590.0 -37.794071  144.926097
6         Melbourne     591.0 -37.814637  144.963924
7   North Melbourne     592.0 -37.799290  144.946400
8         Parkville     593.0 -37.787375  144.952417
9    Port Melbourne     594.0 -37.827094  144.916919
10      South Wharf     840.0 -37.825345  144.952697
11      South Yarra     595.0 -37.835980  144.983190
12        Southbank     596.0 -37.825574  144.963925
13   West Melbourne     597.0 -37.807599  144.941566

In df1 we have:

##### 6	Melbourne	Melbourne 3000
##### 7	Melbourne	Melbourne 3004
    
We simply asume Melbourne as the combination of both for simplicity.

In [52]:
address = 'Melbourne'
geolocator = Nominatim()
location = geolocator.geocode(address)
Latitude = location.latitude
Longitude = location.longitude
print('The geograpical coordinate of Melbourne are {},{}.'.format(Latitude,Longitude))

/opt/conda/envs/Python36/lib/python3.6/site-packages/ipykernel/__main__.py:2: DeprecationWarning: Using Nominatim with the default "geopy/1.18.1" `user_agent` is strongly discouraged, as it violates Nominatim's ToS https://operations.osmfoundation.org/policies/nominatim/ and may possibly cause 403 and 429 HTTP errors. Please specify a custom `user_agent` with `Nominatim(user_agent="my-application")` or by overriding the default `user_agent`: `geopy.geocoders.options.default_user_agent = "my-application"`. In geopy 2.0 this will become an exception.
  from ipykernel import kernelapp as app


The geograpical coordinate of Melbourne are -37.8142176,144.9631608.


In [76]:
# We create a map of Melbourne using latitude and longitude values using Folium
map_Melbourne = folium.Map(location=[Latitude, Longitude], zoom_start=11)

# add markers to map
for lat, lng, label in zip(Melbourne['LATITUDE'], Melbourne['LONGITUDE'], Melbourne['Suburb']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_Melbourne)  
    
map_Melbourne

# NOW THAT WE HAVE THE CITY SEGMENTED BY SUBURB WITH THE RESPECTIVE COORDINATES, I'D LIKE TO KNOW HOUSING PRICES FOR A 3 BEDROOM APARTMENT


In [55]:
url3 = 'https://raw.githubusercontent.com/marianopuga/Battle-of-the-neighborhoods/master/Melbourne_housing_FULL.csv'
read_table = pd.read_csv(url3,header=[0])
df3 = read_table
df3

Suburb              Address  Rooms Type      Price Method  \
0              Abbotsford        68 Studley St      2    h        NaN     SS   
1              Abbotsford         85 Turner St      2    h  1480000.0      S   
2              Abbotsford      25 Bloomburg St      2    h  1035000.0      S   
3              Abbotsford   18/659 Victoria St      3    u        NaN     VB   
4              Abbotsford         5 Charles St      3    h  1465000.0     SP   
5              Abbotsford     40 Federation La      3    h   850000.0     PI   
6              Abbotsford          55a Park St      4    h  1600000.0     VB   
7              Abbotsford         16 Maugie St      4    h        NaN     SN   
8              Abbotsford         53 Turner St      2    h        NaN      S   
9              Abbotsford         99 Turner St      2    h        NaN      S   
10             Abbotsford       129 Charles St      2    h   941000.0      S   
11             Abbotsford         124 Yarra St      3    h  1876000.0      S   
12             Abbotsford  121/56 Nicholson St      2    u        NaN     PI   
13             Abbotsford        17 Raphael St      4    h        NaN      W   
14             Abbotsford        98 Charles St      2    h  1636000.0      S   
15             Abbotsford     217 Langridge St      3    h  1000000.0      S   
16             Abbotsford      18a Mollison St      2    t   745000.0      S   
17             Abbotsford   6/241 Nicholson St      1    u   300000.0      S   
18             Abbotsford        10 Valiant St      2    h  1097000.0      S   
19             Abbotsford  403/609 Victoria St      2    u   542000.0      S   
20             Abbotsford            2 Rich St      2    h        NaN     SP   
21             Abbotsford    25/84 Trenerry Cr      2    u   760000.0     SP   
22             Abbotsford    106/119 Turner St      1    u   481000.0     SP   
23             Abbotsford   411/8 Grosvenor St      2    u   700000.0     VB   
24             Abbotsford      40 Nicholson St      3    h  1350000.0     VB   
25             Abbotsford  123/56 Nicholson St      2    u   750000.0      S   
26             Abbotsford           22 Park St      4    h  1985000.0      S   
27             Abbotsford    13/84 Trenerry Cr      1    u   500000.0      S   
28             Abbotsford        45 William St      2    h  1172500.0      S   
29             Abbotsford   7/20 Abbotsford St      1    u   441000.0     SP   
...                   ...                  ...    ...  ...        ...    ...   
34827      West Footscray         13 Barton St      4    h  1096000.0      S   
34828      West Footscray          30 Gwelo St      3    h   900000.0     PI   
34829      West Melbourne     9/11 Anderson St      3    u        NaN      S   
34830      West Melbourne       122 Railway Pl      3    t  1150000.0      S   
34831         Westmeadows        35 Arnside Cr      4    h   675000.0      S   
34832         Westmeadows        2/8 Forman St      4    u   689000.0      S   
34833       Wheelers Hill     45 Chancellor Dr      4    h   936000.0      S   
34834       Wheelers Hill         112 Jells Rd      4    h        NaN     PN   
34835       Wheelers Hill           20 Mill Ct      3    h        NaN      S   
34836       Wheelers Hill          9 Walden Av      5    h  1400000.0      S   
34837        Williamstown      13 Kingshott Cl      5    h  2020000.0      S   
34838        Williamstown   11/81 Melbourne Rd      2    u   450000.0     SP   
34839        Williamstown     24/18 Station Rd      2    u   475000.0     SP   
34840        Williamstown      3/2 Thompson St      2    u   520000.0     VB   
34841        Williamstown        5 Thompson St      2    h   940000.0     VB   
34842        Williamstown        34 Twyford St      3    h  1400000.0     SP   
34843  Williamstown North          32 Edina St      3    h   830000.0      S   
34844             Windsor     4/16 Lewisham Rd      1    u   440000.0     SP   
34845             W

## DROP UNNECESSARY DATA

In [56]:
df3.drop(['Method','SellerG','Distance','Postcode','Car','BuildingArea','YearBuilt','CouncilArea','Regionname','Propertycount'], axis = 1, inplace = True)
df3

Suburb              Address  Rooms Type      Price  \
0              Abbotsford        68 Studley St      2    h        NaN   
1              Abbotsford         85 Turner St      2    h  1480000.0   
2              Abbotsford      25 Bloomburg St      2    h  1035000.0   
3              Abbotsford   18/659 Victoria St      3    u        NaN   
4              Abbotsford         5 Charles St      3    h  1465000.0   
5              Abbotsford     40 Federation La      3    h   850000.0   
6              Abbotsford          55a Park St      4    h  1600000.0   
7              Abbotsford         16 Maugie St      4    h        NaN   
8              Abbotsford         53 Turner St      2    h        NaN   
9              Abbotsford         99 Turner St      2    h        NaN   
10             Abbotsford       129 Charles St      2    h   941000.0   
11             Abbotsford         124 Yarra St      3    h  1876000.0   
12             Abbotsford  121/56 Nicholson St      2    u        NaN   
13             Abbotsford        17 Raphael St      4    h        NaN   
14             Abbotsford        98 Charles St      2    h  1636000.0   
15             Abbotsford     217 Langridge St      3    h  1000000.0   
16             Abbotsford      18a Mollison St      2    t   745000.0   
17             Abbotsford   6/241 Nicholson St      1    u   300000.0   
18             Abbotsford        10 Valiant St      2    h  1097000.0   
19             Abbotsford  403/609 Victoria St      2    u   542000.0   
20             Abbotsford            2 Rich St      2    h        NaN   
21             Abbotsford    25/84 Trenerry Cr      2    u   760000.0   
22             Abbotsford    106/119 Turner St      1    u   481000.0   
23             Abbotsford   411/8 Grosvenor St      2    u   700000.0   
24             Abbotsford      40 Nicholson St      3    h  1350000.0   
25             Abbotsford  123/56 Nicholson St      2    u   750000.0   
26             Abbotsford           22 Park St      4    h  1985000.0   
27             Abbotsford    13/84 Trenerry Cr      1    u   500000.0   
28             Abbotsford        45 William St      2    h  1172500.0   
29             Abbotsford   7/20 Abbotsford St      1    u   441000.0   
...                   ...                  ...    ...  ...        ...   
34827      West Footscray         13 Barton St      4    h  1096000.0   
34828      West Footscray          30 Gwelo St      3    h   900000.0   
34829      West Melbourne     9/11 Anderson St      3    u        NaN   
34830      West Melbourne       122 Railway Pl      3    t  1150000.0   
34831         Westmeadows        35 Arnside Cr      4    h   675000.0   
34832         Westmeadows        2/8 Forman St      4    u   689000.0   
34833       Wheelers Hill     45 Chancellor Dr      4    h   936000.0   
34834       Wheelers Hill         112 Jells Rd      4    h        NaN   
34835       Wheelers Hill           20 Mill Ct      3    h        NaN   
34836       Wheelers Hill          9 Walden Av      5    h  1400000.0   
34837        Williamstown      13 Kingshott Cl      5    h  2020000.0   
34838        Williamstown   11/81 Melbourne Rd      2    u   450000.0   
34839        Williamstown     24/18 Station Rd      2    u   475000.0   
34840        Williamstown      3/2 Thompson St      2    u   520000.0   
34841        Williamstown        5 Thompson St      2    h   940000.0   
34842        Williamstown        34 Twyford St      3    h  1400000.0   
34843  Williamstown North          32 Edina St      3    h   830000.0   
34844             Windsor     4/16 Lewisham Rd      1    u   440000.0   
34845             Windsor           13 Mary St      4    h        NaN   
34846             Wollert           10 Alma Rd      4    h   630000.0   
34847             Wollert      27 Birchmore Rd      3    h   500000.0   
34848             Wollert        16 Gunther Wy      4    h   621000.0   
34849             Wollert      35 Kingscote Wy      3    h   570000.0   
3485

In [57]:
df3.drop(['Address','Date','Type'], axis = 1, inplace = True)

In [58]:
df3

Suburb  Rooms      Price  Bedroom2  Bathroom  Landsize  \
0              Abbotsford      2        NaN       2.0       1.0     126.0   
1              Abbotsford      2  1480000.0       2.0       1.0     202.0   
2              Abbotsford      2  1035000.0       2.0       1.0     156.0   
3              Abbotsford      3        NaN       3.0       2.0       0.0   
4              Abbotsford      3  1465000.0       3.0       2.0     134.0   
5              Abbotsford      3   850000.0       3.0       2.0      94.0   
6              Abbotsford      4  1600000.0       3.0       1.0     120.0   
7              Abbotsford      4        NaN       3.0       2.0     400.0   
8              Abbotsford      2        NaN       4.0       1.0     201.0   
9              Abbotsford      2        NaN       3.0       2.0     202.0   
10             Abbotsford      2   941000.0       2.0       1.0     181.0   
11             Abbotsford      3  1876000.0       4.0       2.0     245.0   
12             Abbotsford      2        NaN       2.0       2.0    4292.0   
13             Abbotsford      4        NaN       6.0       2.0     230.0   
14             Abbotsford      2  1636000.0       2.0       1.0     256.0   
15             Abbotsford      3  1000000.0       NaN       NaN       NaN   
16             Abbotsford      2   745000.0       NaN       NaN       NaN   
17             Abbotsford      1   300000.0       1.0       1.0       0.0   
18             Abbotsford      2  1097000.0       3.0       1.0     220.0   
19             Abbotsford      2   542000.0       NaN       NaN       NaN   
20             Abbotsford      2        NaN       2.0       1.0     176.0   
21             Abbotsford      2   760000.0       NaN       NaN       NaN   
22             Abbotsford      1   481000.0       NaN       NaN       NaN   
23             Abbotsford      2   700000.0       2.0       2.0       0.0   
24             Abbotsford      3  1350000.0       3.0       2.0     214.0   
25             Abbotsford      2   750000.0       2.0       2.0       0.0   
26             Abbotsford      4  1985000.0       NaN       NaN       NaN   
27             Abbotsford      1   500000.0       NaN       NaN       NaN   
28             Abbotsford      2  1172500.0       2.0       1.0     195.0   
29             Abbotsford      1   441000.0       1.0       1.0       0.0   
...                   ...    ...        ...       ...       ...       ...   
34827      West Footscray      4  1096000.0       4.0       1.0     368.0   
34828      West Footscray      3   900000.0       3.0       1.0     460.0   
34829      West Melbourne      3        NaN       NaN       NaN       NaN   
34830      West Melbourne      3  1150000.0       3.0       2.0     678.0   
34831         Westmeadows      4   675000.0       4.0       2.0     560.0   
34832         Westmeadows      4   689000.0       NaN       NaN       NaN   
34833       Wheelers Hill      4   936000.0       4.0       2.0     770.0   
34834       Wheelers Hill      4        NaN       4.0       2.0     714.0   
34835       Wheelers Hill      3        NaN       3.0       2.0     677.0   
34836       Wheelers Hill      5  1400000.0       5.0       2.0     739.0   
34837        Williamstown      5  2020000.0       5.0       3.0     450.0   
34838        Williamstown      2   450000.0       NaN       NaN       NaN   
34839        Williamstown      2   475000.0       NaN       NaN       NaN   
34840        Williamstown      2   520000.0       NaN       NaN       NaN   
34841        Williamstown      2   940000.0       2.0       2.0     122.0   
34842        Williamstown      3  1400000.0       3.0       2.0     325.0   
34843  Williamstown North      3   830000.0       3.0       1.0     345.0   
34844             Windsor      1   440000.0       NaN       NaN       NaN   
34845             Windsor      4        NaN       4.0       3.0     279.0   
34846             Wollert      4   630000.0       4.0       2.0     388.0   
34847          

In [62]:
Melbourne_houseprices = df3[df3['Bedroom2'] == 3.0].reset_index(drop=True)
Melbourne_houseprices.head()

Suburb  Rooms      Price  Bedroom2  Bathroom  Landsize  Lattitude  \
0  Abbotsford      3        NaN       3.0       2.0       0.0   -37.8114   
1  Abbotsford      3  1465000.0       3.0       2.0     134.0   -37.8093   
2  Abbotsford      3   850000.0       3.0       2.0      94.0   -37.7969   
3  Abbotsford      4  1600000.0       3.0       1.0     120.0   -37.8072   
4  Abbotsford      4        NaN       3.0       2.0     400.0   -37.7965   

   Longtitude  
0    145.0116  
1    144.9944  
2    144.9969  
3    144.9941  
4    144.9965

## WE GROUP BY SUBURB AND GROUP ONLY ACCOMMODATIONS WITH 3 BEDROOMS

In [77]:
Melbourne_houseprices2 = Melbourne_houseprices.groupby('Suburb').mean().reset_index()
Melbourne_houseprices2

Suburb     Rooms         Price  Bedroom2  Bathroom  \
0            Abbotsford  3.000000  1.239323e+06       3.0  1.571429   
1            Aberfeldie  3.080000  1.377773e+06       3.0  1.800000   
2          Airport West  3.040541  8.158768e+05       3.0  1.445946   
3             Albanvale  3.000000  5.324167e+05       3.0  1.454545   
4           Albert Park  3.034483  2.065359e+06       3.0  1.586207   
5                Albion  2.969697  6.733667e+05       3.0  1.303030   
6            Alphington  3.050000  1.461062e+06       3.0  1.500000   
7                Altona  3.017241  8.943352e+05       3.0  1.448276   
8        Altona Meadows  3.000000  6.364091e+05       3.0  1.588235   
9          Altona North  3.016667  8.111509e+05       3.0  1.200000   
10               Ardeer  3.000000  6.196071e+05       3.0  1.263158   
11             Armadale  2.906250  1.649569e+06       3.0  1.718750   
12           Ascot Vale  3.020202  1.116307e+06       3.0  1.535354   
13            Ashburton  2.982759  1.570118e+06       3.0  1.568966   
14              Ashwood  3.048780  1.172870e+06       3.0  1.414634   
15            Aspendale  3.000000  9.930000e+05       3.0  1.600000   
16    Aspendale Gardens  3.000000  8.803214e+05       3.0  1.714286   
17              Attwood  3.000000  6.400000e+05       3.0  2.000000   
18     Avondale Heights  3.012821  7.853309e+05       3.0  1.294872   
19           Avonsleigh  3.000000           NaN       3.0  1.000000   
20        Bacchus Marsh  3.000000  3.823333e+05       3.0  1.666667   
21            Balaclava  3.071429  1.364000e+06       3.0  1.571429   
22               Balwyn  3.060606  1.823077e+06       3.0  1.621212   
23         Balwyn North  3.117117  1.662564e+06       3.0  1.585586   
24            Bayswater  3.058824  7.720385e+05       3.0  1.647059   
25      Bayswater North  3.000000  6.976667e+05       3.0  1.545455   
26         Beaconsfield  3.000000  5.370000e+05       3.0  1.500000   
27   Beaconsfield Upper  3.000000  7.640000e+05       3.0  2.000000   
28            Beaumaris  3.000000  1.684071e+06       3.0  1.916667   
29            Bellfield  3.000000  8.320588e+05       3.0  1.250000   
..                  ...       ...           ...       ...       ...   
295              Toorak  3.070423  1.996536e+06       3.0  1.901408   
296          Travancore  3.333333  1.498500e+06       3.0  2.333333   
297           Truganina  3.000000  5.001062e+05       3.0  2.000000   
298         Tullamarine  3.000000  6.408478e+05       3.0  1.360000   
299             Vermont  3.000000  9.731364e+05       3.0  1.357143   
300       Vermont South  3.000000  9.468333e+05       3.0  1.800000   
301            Viewbank  3.000000  8.836522e+05       3.0  1.548387   
302              Wallan  3.000000  4.162000e+05       3.0  1.300000   
303            Wantirna  3.000000  8.828571e+05       3.0  1.636364   
304      Wantirna South  3.000000  9.101667e+05       3.0  1.521739   
305          Warrandyte  3.000000  9.415000e+05       3.0  1.750000   
306           Waterways  3.000000  6.510000e+05       3.0  2.000000   
307            Watsonia  3.056604  7.783438e+05       3.0  1.283019   
308      Watsonia North  3.000000  7.437273e+05       3.0  1.285714   
309            Werribee  3.010526  5.205867e+05       3.0  1.431579   
310      West Footscray  3.012195  8.804514e+05       3.0  1.195122   
311      West Melbourne  3.000000  1.319632e+06       3.0  1.958333   
312         Westmeadows  3.000000  5.923889e+05       3.0  1.300000   
313       Wheelers Hill  3.000000  1.051157e+06       3.0  1.800000   
314          Whittlesea  3.000000  5.205000e+05       3.0  1.500000   
315    Williams Landing  3.000000  5.150000e+05       3.0  2.000000   
316        Williamstown  3.078652  1.360623e+06       3.0  1.662921   
317  Williamstown North  3.000000  1.018700e+06       3.0  1.600000   
318             Windsor  3.028571  1.642300e+06       3.0  1.657143   
319             Wollert  3.000000  5.4595

In [81]:
Melbourne_houseprices3 = Melbourne_houseprices2.loc[(Melbourne_houseprices2['Suburb'] == 'Carlton')|
                                                   (Melbourne_houseprices2['Suburb'] == 'Carlton North')|
                                                   (Melbourne_houseprices2['Suburb'] == 'East Melbourne')|
                                                   (Melbourne_houseprices2['Suburb'] == 'Flemington')|
                                                   (Melbourne_houseprices2['Suburb'] == 'Kensington')|
                                                   (Melbourne_houseprices2['Suburb'] == 'Melbourne')|
                                                   (Melbourne_houseprices2['Suburb'] == 'North Melbourne')|
                                                   (Melbourne_houseprices2['Suburb'] == 'Parkville')|
                                                   (Melbourne_houseprices2['Suburb'] == 'Port Melbourne')|
                                                   (Melbourne_houseprices2['Suburb'] == 'South Wharf')|
                                                   (Melbourne_houseprices2['Suburb'] == 'South Yarra')|
                                                   (Melbourne_houseprices2['Suburb'] == 'Southbank')|
                                                   (Melbourne_houseprices2['Suburb'] == 'West Melbourne')].reset_index(drop=True) 

                                                   
Melbourne_houseprices3                                                   
                                         

Suburb     Rooms         Price  Bedroom2  Bathroom    Landsize  \
0           Carlton  3.000000  1.238938e+06       3.0  1.428571  898.894737   
1     Carlton North  2.970588  1.661591e+06       3.0  1.294118  220.060606   
2    East Melbourne  3.000000  2.362500e+06       3.0  2.000000   77.500000   
3        Flemington  3.000000  1.099500e+06       3.0  1.241379  238.923077   
4        Kensington  3.020833  1.092933e+06       3.0  1.312500  369.883721   
5         Melbourne  3.000000  1.012000e+06       3.0  2.166667   15.833333   
6   North Melbourne  3.076923  1.348921e+06       3.0  1.769231  217.619048   
7         Parkville  3.000000  1.508909e+06       3.0  1.846154  947.454545   
8    Port Melbourne  3.009804  1.628356e+06       3.0  1.696078  369.285714   
9       South Yarra  3.000000  1.714935e+06       3.0  1.797980  325.988235   
10        Southbank  3.000000  8.280000e+05       3.0  1.444444   24.875000   
11   West Melbourne  3.000000  1.319632e+06       3.0  1.958333  272.263158   

    Lattitude  Longtitude  
0  -37.799670  144.964539  
1  -37.784755  144.973818  
2  -37.815092  144.985514  
3  -37.784436  144.927914  
4  -37.793306  144.926958  
5  -37.810040  144.966160  
6  -37.799070  144.947444  
7  -37.785196  144.956013  
8  -37.836906  144.938268  
9  -37.841587  144.992771  
10 -37.827756  144.965848  
11 -37.808377  144.947529

# NOW WE SORT OUR HOUSING PRICES IN DECENDING ORDER

In [83]:
Melbourne_houseprices4 = Melbourne_houseprices3.sort_values(by=['Price'], ascending=False).reset_index(drop=True) 
Melbourne_houseprices4

Suburb     Rooms         Price  Bedroom2  Bathroom    Landsize  \
0    East Melbourne  3.000000  2.362500e+06       3.0  2.000000   77.500000   
1       South Yarra  3.000000  1.714935e+06       3.0  1.797980  325.988235   
2     Carlton North  2.970588  1.661591e+06       3.0  1.294118  220.060606   
3    Port Melbourne  3.009804  1.628356e+06       3.0  1.696078  369.285714   
4         Parkville  3.000000  1.508909e+06       3.0  1.846154  947.454545   
5   North Melbourne  3.076923  1.348921e+06       3.0  1.769231  217.619048   
6    West Melbourne  3.000000  1.319632e+06       3.0  1.958333  272.263158   
7           Carlton  3.000000  1.238938e+06       3.0  1.428571  898.894737   
8        Flemington  3.000000  1.099500e+06       3.0  1.241379  238.923077   
9        Kensington  3.020833  1.092933e+06       3.0  1.312500  369.883721   
10        Melbourne  3.000000  1.012000e+06       3.0  2.166667   15.833333   
11        Southbank  3.000000  8.280000e+05       3.0  1.444444   24.875000   

    Lattitude  Longtitude  
0  -37.815092  144.985514  
1  -37.841587  144.992771  
2  -37.784755  144.973818  
3  -37.836906  144.938268  
4  -37.785196  144.956013  
5  -37.799070  144.947444  
6  -37.808377  144.947529  
7  -37.799670  144.964539  
8  -37.784436  144.927914  
9  -37.793306  144.926958  
10 -37.810040  144.966160  
11 -37.827756  144.965848

# With this we end our data exploration
.........................................................................................................................................




# Coursera Capstone Project



## The Battle of Neighborhoods (Week 2)



### Mariano Puga - Part 2 : METHODOLOGY, RESULTS, DISCUSSION AND CONCLUSION
































# 3. Methodology section

### NOW WE DEFINE THE FORESQUARE CREDENTIALS AND VERSION WE ARE USING

In [44]:
CLIENT_ID = '0GGVWFVY53QQXPAFMF5EBLBMGTEV01Q1Q121HZESIB2ETE1H' # your Foursquare ID
CLIENT_SECRET = 'TGGYTLBAT13J2PRP15P1E0MBAKMETADT2OYUU3KO2C4U5ZNL' # your Foursquare Secret
VERSION = '20200202' # Foursquare API version

### BACK TO MELBOURNE, WE GET THE TOP 100 VENUES WITHIN A 500 MTS RADIUS

In [45]:
LIMIT = 100
radius = 500

In [46]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url4 = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url4).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Suburb', 
                  'Suburb Latitude', 
                  'Suburb Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category' ]
    
    return(nearby_venues)

In [47]:
Melbourne_venues = getNearbyVenues(names=Melbourne['Suburb'],
                                   latitudes=Melbourne['LATITUDE'],
                                   longitudes=Melbourne['LONGITUDE']
                                    )

South Yarra


KeyError: 'groups'

In [ ]:
print(Melbourne_venues.shape)
Melbourne_venues.head()

In [ ]:
print('There are {} uniques categories.'.format(len(Melbourne_venues['Venue Category'].unique())))

#### WE ANALYZE THE SUBURBS, WE CREATE DUMMIES IN ORDER TO CHECK WHICH TYPE OF VENUES WE HAVE

In [ ]:
# one hot encoding
Melbourne_onehot = pd.get_dummies(Melbourne_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
Melbourne_onehot['Suburb'] = Melbourne_venues['Suburb'] 

# move neighborhood column to the first column
fixed_columns = [Melbourne_onehot.columns[-1]] + list(Melbourne_onehot.columns[:-1])
Melbourne_onehot = Melbourne_onehot[fixed_columns]

Melbourne_onehot.head()

In [ ]:
#RESULTS

In [ ]:
#DISCUSSION

In [ ]:
#CONCLUSION